design-proposals/klip-7-connect-integration.md

```diff
@@ -0,0 +1,224 @@
+      ...);
+
+--------------
+SOURCE CREATED
+--------------
+```
+
+If the connector is one of the KSQL supported connectors, we can have default configuration options
+and tighter integration specified by `CREATE EXTERNAL SOURCE(type='JDBC') WITH (...);`.
+
+*Note: as Connect manages to simplify the configuration, whether through templates or any other
+means, we will immediately be able to leverage this simplification in KSQL's WITH clause.*
+
+Next, we provide tools to understand the status of the connector and corresponding topics hitting
+the `GET /connectors` and `GET /connectors/<name>` REST endpoints:
+```sql
+ksql> SHOW EXTERNAL SOURCES;
+
+    name         |              connector                        |   topic prefix
+-----------------|-----------------------------------------------|--------------------
+my-postgres-jdbc | io.confluent.connect.jdbc.JdbcSourceConnector | test-postgres-jdbc
+
+ksql> DESCRIBE my-postgres-jdbc;
+
+name: my-postgres-jdbc
+connector: io.confluent.connect.jdbc.SourceConnector
+topic-prefix: test-postgres-jdbc
+
+   worder id     | task | status
+-----------------|------|---------
+10.200.7.69:8083 | 0    | RUNNING
+
+ name      |     topic                     |  schema
+-----------|-------------------------------|--------------------------------------
+user-table | test-postgres-jdbc-user-table | id VARCHAR, user VARCHAR, age INTEGER
+page-table | test-postgres-jdbc-page-table | id VARCHAR, page VARCHAR, url VARCHAR
+```
+
+Then the user can create streams or tables from the underlying topics (this can happen automatically
+for some set of blessed connectors):
+```sql
+ksql> CREATE TABLE users_original WITH (kafka_topic = 'test-postgres-jdbc-user-table', ...);
+
+--------------
+STREAM CREATED
+--------------
+```
+
+At this point, assuming the `pageviews_original` stream is already created, we can join the newly
+created `users_original` table with the `pageviews_original` stream. Finally, the user can drop and 
+terminate the connectors.
+```sql
+ksql> DROP EXTERNAL SOURCE my-postgres-jdbc;
+```
+
+**Sample Egress**
+
+Data egress would work in a similar fashion:
+```sql
+ksql> CREATE EXTERNAL SINK my-elastic-sink WITH (sources='my-table1', ...);
+
+ksql> SHOW EXTERNAL SINKS;
+
+    name         |              connector                        |   topic prefix
+-----------------|-----------------------------------------------|--------------------
+my-elastic-sink  | io.confluent.connect.elasticsearch.ElasticsearchSinkConnector | elastic-sink
+
+ksql> DESCRIBE my-elastic-sink;
+
+name: my-elastic-sink
+connector: io.confluent.connect.elasticsearch.ElasticsearchSinkConnector
+topic-prefix: elstic-sink
+
+   worder id     | task | status
+-----------------|------|---------
+10.200.7.69:8083 | 1    | RUNNING
+
+name           | topic
+--------------------------------------------
+elasticsearch1 | elastic-sink-elasticsearch1 
+
+ksql> DROP EXTERNAL SINK my-elastic-sink TERMINATE;
```

Comment:
Isn't `CASCADE` a more standard name databases use?

Reply:
Oops haven't updated this - I'm going to remove the `TERMINATE` option entirely